src/dict.c

```diff
@@ -964,11 +974,13 @@ static int _dictExpandIfNeeded(dict *d)
      * table (global setting) or we should avoid it but the ratio between
      * elements/buckets is over the "safe" threshold, we resize doubling
      * the number of buckets. */
+    unsigned long new_size = d->ht[0].used * 2;
```

Comment:
AFAIK, in most case, we expand dict when `ht[0].used` is equal to `d->ht[0].size`. do you mean if `d->ht[0].used > d->ht[0].size`, we can use `ht[0].used` rather than `ht[0].used * 2` to try again?

Reply:
from what i understand is that when it decides to rehash (when used > size), it decides to go for used*2 so that it doesn't have to rehash soon again.
so when the server is near it's limit, and *2 gets rejected, we can attempt without *2, maybe that one will be ok (not cause eviction)

Reply:
@oranagra I did that, please have a look

Reply:
seems good. still waiting for feedback on the other open issues. i'll try to shake the tree.